plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1161,44 +1165,78 @@ public void setTableProperties(ConnectorSession session, ConnectorTableHandle ta
         IcebergTableHandle table = (IcebergTableHandle) tableHandle;
         BaseTable icebergTable = (BaseTable) catalog.loadTable(session, table.getSchemaTableName());
 
+        Set<String> unsupportedProperties = Sets.difference(properties.keySet(), UPDATABLE_PROPERTIES);
+        if (!unsupportedProperties.isEmpty()) {
+            throw new TrinoException(NOT_SUPPORTED, "The following properties cannot be updated: " + String.join(", ", unsupportedProperties));
+        }
+
         transaction = icebergTable.newTransaction();
-        UpdateProperties updateProperties = transaction.updateProperties();
 
-        for (Map.Entry<String, Optional<Object>> propertyEntry : properties.entrySet()) {
-            String trinoPropertyName = propertyEntry.getKey();
-            Optional<Object> propertyValue = propertyEntry.getValue();
+        if (properties.containsKey(FILE_FORMAT_PROPERTY) || properties.containsKey(FORMAT_VERSION_PROPERTY)) {
+            UpdateProperties updateProperties = transaction.updateProperties();
 
-            switch (trinoPropertyName) {
-                case FILE_FORMAT_PROPERTY:
-                    updateProperties.defaultFormat(((IcebergFileFormat) propertyValue.orElseThrow()).toIceberg());
-                    break;
-                case FORMAT_VERSION_PROPERTY:
-                    // UpdateProperties#commit will trigger any necessary metadata updates required for the new spec version
-                    updateProperty(updateProperties, FORMAT_VERSION, propertyValue, formatVersion -> Integer.toString((int) formatVersion));
-                    break;
-                default:
-                    // TODO: Support updating partitioning https://github.com/trinodb/trino/issues/12174
-                    throw new TrinoException(NOT_SUPPORTED, "Updating the " + trinoPropertyName + " property is not supported");
+            Optional<Object> fileFormat = properties.getOrDefault(FILE_FORMAT_PROPERTY, Optional.empty());
+            if (fileFormat.isPresent()) {
+                updateProperties.defaultFormat(((IcebergFileFormat) fileFormat.orElseThrow()).toIceberg());
+            }
+
+            Optional<Object> formatVersion = properties.getOrDefault(FORMAT_VERSION_PROPERTY, Optional.empty());
+            if (formatVersion.isPresent()) {
+                // UpdateProperties#commit will trigger any necessary metadata updates required for the new spec version
+                updateProperties.set(FORMAT_VERSION, Integer.toString((int) formatVersion.orElseThrow()));
```

Comment:
I think in such limited scope it's fine to `map.get` + `!= null`.
While i value desire not to use nulls, `map.getOrDefault(k, Optional.empty())` isn't idiomatic (yet)

Reply:
Sure, will update. I'm not a fan of `optionalThing != null && optionalThing.isPresent()` but that's okay

Reply:
Oh, right. i mis-read this. map values are `Optional`. 
However, `Optional.empty()`  value means "reset to default", while no entry means "no change". Am i right?